src/NumericalAlgorithms/SphericalHarmonic/ComplexDataSlice.hpp

```diff
@@ -0,0 +1,171 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <complex>
+#include <cstddef>
+
+#include "DataStructures/ComplexDataVector.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/Gsl.hpp"
+
+enum ComplexSliceKind {
+  interleaved,
+  real_slices,
+};
+
+/// \brief A storage container for storing sub-vectors ("slices") of a
```

Comment:
After having looked at more of the code I get more confused about what problem this class is trying to solve. Maybe we could talk about this at the end of the Monday call. It seems like it's sort of a view, but not really but sometimes yes? Except it also lets you change representation of the data. Maybe the cleaner thing is to have a view and a function that does the translation. Not sure.

Reply:
This is primarily intended for use with libsharp, which takes pointers to real and imaginary parts, as well as strides. So, it's possible to either pass it pointers to contiguous blocks of real and imaginary parts separately, or just point it into a full complex block and have twice the stride. 
It's not clear which will be more efficient in realistic scenarios, and could easily depend on e.g. which spin is being evaluated. So, I want a switch to choose between those two scenarios, and a wrapper class to handle all of the memory management that doesn't really need to be in the (already complicated) code which calls the libsharp execution.

Sorry if this PR makes those goals unclear. I'm happy to consider other options if you think those can be done with less fooling about than is here.